homeassistant/components/nextbus/sensor.py

```diff
@@ -103,12 +100,13 @@ class NextBusDepartureSensor(SensorEntity):
     _attr_device_class = SensorDeviceClass.TIMESTAMP
     _attr_icon = "mdi:bus"
 
-    def __init__(self, client, agency, route, stop, name=None):
+    def __init__(self, client, unique_id, agency, route, stop, name=None):
         """Initialize sensor with all required config."""
         self.agency = agency
         self.route = route
         self.stop = stop
         self._attr_extra_state_attributes = {}
+        self._attr_unique_id = unique_id
 
         # Maybe pull a more user friendly name from the API here
         self._attr_name = f"{agency} {route}"
```

Comment:
```suggestion
        self._attr_name = name
```

Please delete the following as the name is always set:
```
if name:
            self._attr_name = name
```